main.tf

```diff
@@ -1,152 +1,165 @@
+  bastion_user             = var.bastion_user
+  bastion_private_key_path = var.enable_bastion || var.bastion_public_ip != "" ? var.bastion_private_key_path : ""
+  compute_user             = var.compute_user
+  freeform_tags            = var.freeform_tags
 }
 
-# configuration
-module "olcne" {
-  source = "./modules/olcne"
-
-  # general oci parameters
-  compartment_id = var.compartment_id
-  label_prefix   = var.label_prefix
-  region         = var.region
-
-  # bastion
-  bastion_public_ip    = module.base.bastion_public_ip
-  ssh_private_key_path = var.ssh_private_key_path
-
-  # operator
-  operator_ip = module.base.operator_private_ip
-
-  # list of master nodes
-  olcne_masters = local.olcne_masters
-
-  # private CA Certificate
-  olcne_certificate = local.olcne_certificate
-
-  # list of worker nodes
-  olcne_workers = local.olcne_workers
-
-  # private key parameter to access other nodes
-  secret_id = var.secret_id
-
-  # olcne environment
-  olcne_environment = local.olcne_environment
-
-  helm_version = var.helm_version
-
-  loadbalancer_ip_address = module.loadbalancer.pub_lb_ip
-  deploy_nginx_software = var.deploy_nginx_software
+module "ocne-provision" {
+  depends_on = [module.infrastructure]
+  source     = "./modules/ocne-provision"
+  count      = var.provision_mode == lookup(local.provision_modes_map,"provision_mode_ocne","") ? 1 : 0
+
+  ssh_public_key_path       = var.ssh_public_key_path
+  ssh_private_key_path      = var.ssh_private_key_path
+  os_version                = var.os_version
+  proxy                     = var.proxy
+  no_proxy                  = var.no_proxy
+  use_vault                 = var.use_vault
+  control_plane_node_count  = var.control_plane_node_count
+  worker_node_count         = var.worker_node_count
+  control_plane_nodes       = module.infrastructure.control_plane_nodes
+  worker_nodes              = module.infrastructure.worker_nodes
+  apiserver_ip              = module.infrastructure.apiserver_ip
+  vault_uri                 = var.use_vault ? module.vault[0].uri : ""
+  ocne_version              = data.external.ocne_config.result.version
+  environment_name          = data.external.ocne_config.result.environment_name
+  kubernetes_name           = data.external.ocne_config.result.kubernetes_name
+  kube_apiserver_port       = var.kube_apiserver_port
+  kube_apiserver_ip         = var.virtual_ip ? module.infrastructure.kube_apiserver_virtual_ip : module.infrastructure.load_balancer_ip
+  virtual_ip                = var.virtual_ip
+  container_registry        = var.container_registry
+  certificate_signing_token = var.use_vault ? module.vault[0].vault_ocne_client_token : ""
+  enable_bastion           = var.enable_bastion
+  bastion_public_ip         = var.enable_bastion ? module.oci-ocne-network[0].bastion_public_ip : var.bastion_public_ip
+  bastion_user              = var.bastion_user
```

Comment:
Do you intend to use other OS e.g. ubuntu for the bastion host? If not, you can keep this to opc